main/src/cgeo/geocaching/connector/al/ALApi.java

```diff
@@ -377,12 +379,21 @@ private static Geocache parseCacheDetail(final JsonNode response) {
             cache.setCoords(new Geopoint(location.get(LATITUDE).asText(), location.get(LONGITUDE).asText()));
             cache.setType(ADVLAB);
             cache.setSize(CacheSize.getById("virtual"));
+            cache.setVotes(response.get("RatingsTotalCount").asInt());
+            cache.setRating(response.get("RatingsAverage").floatValue());
             // cache.setArchived(response.get("IsArchived").asBoolean()); as soon as we're using active mode
             // cache.setFound(response.get("IsComplete").asBoolean()); as soon as we're using active mode
             cache.setDisabled(false);
             cache.setHidden(parseDate(response.get("PublishedUtc").asText()));
             cache.setOwnerDisplayName(response.get("OwnerUsername").asText());
             cache.setWaypoints(parseWaypoints((ArrayNode) response.path("GeocacheSummaries")), true);
+            final boolean isLinear = response.get("IsLinear").asBoolean();
+            if (isLinear) {
+                cache.setAlcMode(1);
```

Comment:
What about using an `Enum` instead of magic numbers? The `Enum` could hold the user displayable strings as well, so no additional mapping would be necessary...

Reply:
sure, could be done, however, I had actually envisioned to turn this into bitfield operations if, and only if more modes are needed. I was looking at: https://docs.oracle.com/javase/8/docs/api/java/util/BitSet.html

To instantiate: `BitSet bitField = new BitSet(); //can specify size`

To check if bit n is true `bitField.get(n);`
To set bit n `bitField.set(n, value); //can also use set(int) and clear(int) instead`

I had hoped, I could later, once needed and if needed at all cast the `alcMode` which is an integer now into a `BitSet` object. The main reason for this would be to avoid adding more columns to the database schema.

Reply:
Hm, don't really understand why we should use a BitSet.

I had thought of an enum with two parameters, `int id` and `@StringRes int userDisplayableName`. You can still store the id value as int inside the DB. However it's IMHO important that only one class needs to remember the actual mapping. 

We have similar code patterns at different places in cgeo, I will search for a meanful example...

Reply:
I will merge now anyway to start preparing a beta version. Switch from "magic numbers" to "enum" could be added later without causing harm.